tools/SourceKit/lib/SwiftLang/SwiftLangSupport.cpp

```diff
@@ -623,10 +623,47 @@ Optional<UIdent> SwiftLangSupport::getUIDForDeclAttribute(const swift::DeclAttri
         return Attr_Objc;
       }
     }
+    case DAK_AccessControl: {
+      static UIdent Attr_Private("source.decl.attribute.private");
+      static UIdent Attr_FilePrivate("source.decl.attribute.fileprivate");
+      static UIdent Attr_Internal("source.decl.attribute.internal");
+      static UIdent Attr_Public("source.decl.attribute.public");
+      static UIdent Attr_Open("source.decl.attribute.open");
+
+      switch (cast<AbstractAccessControlAttr>(Attr)->getAccess()) {
+        case AccessLevel::Private:
+          return Attr_Private;
+        case AccessLevel::FilePrivate:
+          return Attr_FilePrivate;
+        case AccessLevel::Internal:
+          return Attr_Internal;
+        case AccessLevel::Public:
+          return Attr_Public;
+        case AccessLevel::Open:
+          return Attr_Open;
+      }
+    }
+    case DAK_SetterAccess: {
+      static UIdent Attr_Private("source.decl.attribute.setter_access.private");
+      static UIdent Attr_FilePrivate("source.decl.attribute.setter_access.fileprivate");
+      static UIdent Attr_Internal("source.decl.attribute.setter_access.internal");
+      static UIdent Attr_Public("source.decl.attribute.setter_access.public");
+      static UIdent Attr_Open("source.decl.attribute.setter_access.open");
+
+      switch (cast<AbstractAccessControlAttr>(Attr)->getAccess()) {
```

Comment:
Is there a way we can share the two switch statements? It seems to me a logic duplication.

Reply:
I agree that there's some logic duplication, but I can't think of a way of sharing those and being easier to read at the same time.